src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/KeyValuePairConverter.cs

```diff
@@ -3,23 +3,55 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Text.Encodings.Web;
 
 namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class KeyValuePairConverter<TKey, TValue> : JsonValueConverter<KeyValuePair<TKey, TValue>>
     {
-        private const string KeyName = "Key";
-        private const string ValueName = "Value";
+        private const string KeyNameCLR = "Key";
+        private const string ValueNameCLR = "Value";
 
-        // todo: https://github.com/dotnet/runtime/issues/1197
-        // move these to JsonSerializerOptions and use the proper encoding.
-        private static readonly JsonEncodedText _keyName = JsonEncodedText.Encode(KeyName, encoder: null);
-        private static readonly JsonEncodedText _valueName = JsonEncodedText.Encode(ValueName, encoder: null);
+        // Property name for "Key" and "Value" with Options.PropertyNamingPolicy applied.
+        private string _keyName = null!;
+        private string _valueName = null!;
+
+        // _keyName and _valueName as JsonEncodedText.
+        private JsonEncodedText _keyNameEncoded;
```

Comment:
We can cache these on this class as above (and don't have to pair to the options class) only because this type is internal and a new instance is created for every options instance. If this was public and\or we expected different option instances to be passed to read\write then we'd have to pair it to the options instance.

Reply:
Yes, I have an iteration that caches them on the options instance, but this way is bit cleaner for now (at the cost of doing the conversions for every KeyValuePair<,> type in the object graph, rather than once (lazily) for each options instance).